### PR TITLE
fix: add PathUtil class + get_base_path

### PR DESCRIPTION
**Pull request by ThomasvLingen**

### CODE REVIEW

File: src/engine/PathUtil.cpp [hunk 1]
```diff
@@ -0,0 +1,28 @@
+//
+// Created by mafn on 11/29/16.
+//
+
+#include <SDL2/SDL.h>
+#include "PathUtil.hpp"
+
+namespace Engine {
+
+    bool PathUtil::_initialised = false;
```
Comment (jorgdebont) [inline src/engine/PathUtil.cpp]: Please place this inside of the function

Comment (jorgdebont) [inline src/engine/PathUtil.cpp]: is never set to true

File: src/engine/PathUtil.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    string PathUtil::_run_path = "";
```
Comment (jorgdebont) [inline src/engine/PathUtil.cpp]: you can also check if the run_path isnt empty instead of using a boolean

**Comment by ThomasvLingen** [comment]
On second thought, this is retarded code

File: src/engine/PathUtil.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    void PathUtil::_init_run_path()
+    {
+        char* SDL_base_path = SDL_GetBasePath();
+        PathUtil::_run_path = string(SDL_base_path);
+        SDL_free(SDL_base_path);
+    }
+
+    string PathUtil::get_run_path()
```
Comment (jorgdebont) [inline src/engine/PathUtil.cpp]: Perhaps use a define, like BASEPATH

File: src/gamelogic/state/LoadState.cpp [hunk 4]
```diff
@@ -19,54 +19,54 @@ namespace State {
         this->_context.SDL_facade.clear_screen();
         this->_context.SDL_facade.draw_rect({0,0}, 640, 480, Color{0,0,0});
 
-        this->_loading_screen_time = this->_loading_screen_time + 2; //TODO: To ticks
+        this->_loading_screen_time = this->_loading_screen_time + 20; //TODO: To ticks
```
Comment (jorgdebont) [inline src/gamelogic/state/LoadState.cpp]: unrelated change

File: src/engine/PathUtil.cpp (hunk 3 above)
Comment (jorgdebont) [inline src/engine/PathUtil.cpp]: also makes sure that you dont have to place using Engine::PathUtil everywhere

File: src/engine/PathUtil.cpp (hunk 1 above)
Comment (ThomasvLingen) [inline src/engine/PathUtil.cpp]: done

Comment (ThomasvLingen) [inline src/engine/PathUtil.cpp]: no longer there now

File: src/engine/PathUtil.cpp (hunk 2 above)
Comment (ThomasvLingen) [inline src/engine/PathUtil.cpp]: done

File: src/engine/PathUtil.cpp (hunk 3 above)
Comment (ThomasvLingen) [inline src/engine/PathUtil.cpp]: we `VIDYA_RUNPATH` now

File: src/gamelogic/state/LoadState.cpp (hunk 4 above)
Comment (ThomasvLingen) [inline src/gamelogic/state/LoadState.cpp]: Whoops, removed!
